docs/content/en/docs/Installation/p2p.md

```diff
@@ -14,68 +14,215 @@ Feedback and bug reports are welcome, as we are improving the p2p aspects of Kai
+   # A master cluster init is always required,
+   # Any additional master_node is configured as part of the 
+   # HA control plane.
+   # If auto is disabled, HA has no effect.
+   ha:
+     # Enables HA controlplane
+     enable: true
+     # number of HA master node (beside the one used for init) for the control-plane
+     master_nodes: 2
+     # Use an External database for the HA control plane
+     external_db: "external-db-string"
+ # network_token is the shared secret used by the nodes to co-ordinate with p2p
+ network_token: "YOUR_TOKEN_GOES_HERE"
 ```
 
-{{% alert title="Note" %}}
+In the YAML configuration example, there are several important keywords that control the behavior of the automatic cluster deployment:
 
-The `k3s` block can still be used to override other `k3s` settings, e.g. `args`.
+| Keyword | Description |
+| --- | --- |
+| `p2p` | Configures the peer to peer networking of the cluster |
+| `p2p.disable_dht` | Disables the distributed hash table for cluster discovery |
+| `p2p.network_token` | The shared secret used by the nodes to coordinate with p2p |
+| `p2p.network_id` | Optional, unique identifier for the kubernetes cluster. It allows bootstrapping of multiple cluster using the same network token |
+| `p2p.role` | Force a specific role for the node of the cluster |
+| `p2p.vpn` | Configures a VPN for the cluster nodes |
+| `p2p.vpn.create` | Enables the creation of the VPN |
+| `p2p.vpn.use` | Enables the use of the VPN for routing Kubernetes traffic of the cluster |
+| `p2p.vpn.env` | Configures the environment variables used to start for the VPN |
+| `p2p.vpn.auto` | Configures the automatic deployment of the cluster |
+| `p2p.auto.enable` | Enables automatic node configuration for role assignment |
+| `p2p.auto.ha` | Configures the high availability settings for the cluster |
+| `p2p.auto.ha.enable` | Enables the high availability settings |
+| `p2p.auto.ha.master_nodes` | The number of expected HA master nodes in the cluster besides the cluster-init master. |
+| `p2p.auto.ha.external_db` | The external database used for high availability |
 
-{{% /alert %}}
+## Elastic IP
+
+If deploying a cluster in a Local network, it might be preferable to disable the VPN functionalities.
+
+We use KubeVIP to provide an elastic ip for the control plane that can be configured via a specific block:
+
+```yaml
 
+p2p:
+ network_token: ".."
+ vpn:
+   # Disable VPN, so traffic is not configured with a VPN
+   create: false
+   use: false
+
+## Sets the Elastic IP used in KubeVIP
+kubevip:
+  eip: "192.168.1.110"
+  # Specify a manifest URL for KubeVIP. Empty uses default
+  manifest_url: ""
+  # Enables KubeVIP
+  enable: true
+  # Specifies a KubeVIP Interface
+  interface: "ens18"
+```
+
+
+| Keyword | Description |
+| --- | --- |
+| `kubevip` | Block to configure KubeVIP for the cluster |
+| `kubevip.eip` | The Elastic IP used for KubeVIP. Specifying one automatically enables KubeVIP |
```

Comment:
Maybe we should explain that they should pick a free IP address in their network. One that won't collide with any other machine or DHCP.